src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -399,13 +400,31 @@ ErrorIfQueryNotSupported(Query *queryTree)
 		errorHint = filterHint;
 	}
 
-	if (queryTree->setOperations)
+	if (queryTree->setOperations && !subqueryPushdown)
```

Comment:
I think we generally do not prefer to pass a boolean variable to a function and act according to that variable. Do you see an easy way of not doing it?